packages/core/src/ide/process-utils.ts

```diff
@@ -7,56 +7,109 @@
 
     try {
-      // Use wmic for Windows
       if (platform === 'win32') {
-        const command = `wmic process where "ProcessId=${currentPid}" get ParentProcessId /value`;
+        const command = `wmic process where "ProcessId=${currentPid}" get Name,ParentProcessId /value`;
         const { stdout } = await execAsync(command);
-        const match = stdout.match(/ParentProcessId=(\d+)/);
-        parentPid = match ? parseInt(match[1], 10) : 0; // Top of the tree is 0
-      }
-      // Use ps for macOS, Linux, and other Unix-like systems
-      else {
-        const command = `ps -o ppid= -p ${currentPid}`;
+        const nameMatch = stdout.match(/Name=([^\n]*)/);
+        processName = nameMatch ? nameMatch[1].trim() : '';
+        const ppidMatch = stdout.match(/ParentProcessId=(\d+)/);
+        parentPid = ppidMatch ? parseInt(ppidMatch[1], 10) : 0;
+      } else {
+        const command = `ps -o ppid=,command= -p ${currentPid}`;
         const { stdout } = await execAsync(command);
-        const ppid = parseInt(stdout.trim(), 10);
-        parentPid = isNaN(ppid) ? 1 : ppid; // Top of the tree is 1
+        const trimmedStdout = stdout.trim();
+        const ppidString = trimmedStdout.split(/\s+/)[0];
+        const ppid = parseInt(ppidString, 10);
+        parentPid = isNaN(ppid) ? 1 : ppid;
+        const fullCommand = trimmedStdout.substring(ppidString.length).trim();
+        processName = path.basename(fullCommand.split(' ')[0]);
       }
     } catch (_) {
       // This can happen if a process in the chain dies during execution.
       // We'll break the loop and return the last valid PID we found.
       break;
     }
 
+    const isShell = shellAllowlist.some((shell) =>
+      platform === 'win32'
+        ? processName.toLowerCase() === shell.toLowerCase()
+        : processName === shell,
+    );
+
+    if (isShell) {
+      let idePid = parentPid;
+      if (os.platform() === 'linux') {
+        try {
+          const { stdout: cmdOut } = await execAsync(
+            `ps -o command= -p ${idePid}`,
+          );
+          // Check if it's a utility process
+          if (cmdOut.includes('--type=')) {
+            const { stdout: ppidOut } = await execAsync(
+              `ps -o ppid= -p ${idePid}`,
+            );
+            const grandParentPid = parseInt(ppidOut.trim(), 10);
+            if (!isNaN(grandParentPid) && grandParentPid > 1) {
+              idePid = grandParentPid;
+            }
+          }
+        } catch (_) {
+          // Ignore if ps fails, we'll just use the parent pid.
+        }
+      }
+      return idePid;
+    }
+
     // Define the root PID for the current OS
     const rootPid = platform === 'win32' ? 0 : 1;
-
     // If the parent is the root process or invalid, we've found our target.
     if (parentPid === rootPid || parentPid <= 0) {
       break;
     }
     // Move one level up the tree for the next iteration.
     currentPid = parentPid;
   }
+  logger.error(
```

Comment:
is there a. reason we don't just call 
console.error directly?
Generally that aligns with the rest of the codebase.